vendor/bat-native-ads/src/bat/ads/internal/account/account.cc

```diff
@@ -64,27 +64,24 @@ void Account::OnPrefChanged(const std::string& path) {
   }
 }
 
-bool Account::SetWallet(const std::string& id, const std::string& seed) {
-  const WalletInfo& last_wallet = wallet_->Get();
+void Account::SetWallet(const std::string& id, const std::string& seed) {
+  const WalletInfo last_wallet = GetWallet();
 
   if (!wallet_->Set(id, seed)) {
     NotifyInvalidWallet();
-    return false;
+    return;
   }
 
-  const WalletInfo& wallet = wallet_->Get();
+  const WalletInfo wallet = GetWallet();
 
-  if (last_wallet.IsValid() && last_wallet != wallet) {
-    NotifyWalletDidChange(wallet);
+  NotifyWalletDidUpdate(wallet);
 
-    Reset();
+  if (wallet.HasChanged(last_wallet)) {
+    WalletDidChange(wallet);
+    return;
   }
 
-  NotifyWalletDidUpdate(wallet);
-
   TopUpUnblindedTokens();
-
-  return true;
 }
 
 WalletInfo Account::GetWallet() const {
```

Comment:
this doesn't match member function declaration

Reply:
Fixed